pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,172 @@
+)
+
+// Add creates a new ClusterLogging Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	if err := configv1.Install(mgr.GetScheme()); err != nil {
+		return &ReconcileProxyConfig{}
+	}
+
+	return &ReconcileProxyConfig{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("proxyconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// We only care about a configmap source with a specific name/namespace,
+	// so filter events before they are provided to the controller event handlers.
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return handleConfigMap(e.MetaNew)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	if err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred); err != nil {
+		return err
+	}
+
+	// Watch for changes to the proxy resource.
+	if err = c.Watch(&source.Kind{Type: &configv1.Proxy{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// handleConfigMap returns true if meta namespace is "openshift-config".
+// "openshift-config" is the namespace for one or more
+// ConfigMaps that contain user provided trusted CA bundles.
+func handleConfigMap(meta metav1.Object) bool {
+	return meta.GetNamespace() == "openshift-config"
+}
+
+var _ reconcile.Reconciler = &ReconcileProxyConfig{}
+
+// ReconcileProxyConfig reconciles a ClusterLogging object
+type ReconcileProxyConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a cluslter-scoped
+// named "cluster" or a configmap object in namespace "openshift-config".
+func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	loggingNamespacedName := types.NamespacedName{Name: singletonName, Namespace: openshiftNS}
+	proxyNamespacedName := types.NamespacedName{Name: proxyName}
+	if request.NamespacedName == proxyNamespacedName {
+		proxyConfig := &configv1.Proxy{}
+		logrus.Debugf("Proxyconfig reconcile request.Name: '%s'", request.Name)
```

Comment:
This debug doesn't really get us much since we require the proxy name to be an expected one... if we keep this can we change it to be something like `Reconciling for a proxy config event` ?

Reply:
Ok.  Removed.